features/api/services/service_plans.feature

```diff
@@ -15,6 +15,13 @@ Feature: Service plans index page
       Given a product
       Then an admin is able to see its service plans
 
+    Scenario: Service plans are scoped by service
+      Given a product
+      And there are other services with service plans
+      And the product has multiple service plans
+      When an admin is on the service plans page of the product
+      Then only service plans of the product are listed
+
```

Comment:
I think this is better tested in a unit test.

Reply:
An integration test you mean? Or unit test of the controller, however you want to call it.

Yes, that's what I started doing, but then I realized that the page initially has just a single `div`, the actual plans table is rendered via JavaScript

https://github.com/3scale/porta/blob/master/app/views/api/service_plans/index.html.slim

https://github.com/3scale/porta/blob/master/app/views/shared/plans/_plans_index_main_section.html.slim

At most we could test (in an integration test) that the JSON injected in the div's data is correct, but I think it's a big hacky. I preferred to do a full UI test for that.

Reply:
I meant a unit test for `Api::ServicePlansPresenter`. As you said, presenters are not the place to run queries, but since the query is there, it should be tested there, IMO.

Reply:
Ah, gotcha. Yes, makes sense.

Reply:
Added here: https://github.com/3scale/porta/pull/3479/commits/07da04365be28fc4b3a4747694dee8035e1d9b19